src/hotspot/share/memory/guardedMemory.cpp

```diff
@@ -33,7 +33,9 @@ void* GuardedMemory::wrap_copy(const void* ptr, const size_t len, const void* ta
   if (outerp != NULL) {
     GuardedMemory guarded(outerp, len, tag);
     void* innerp = guarded.get_user_ptr();
-    memcpy(innerp, ptr, len);
+    if (ptr != nullptr) {
+      memcpy(innerp, ptr, len);
+    }
```

Comment:
Hmm. innerp can never be null. If anything, we should assert.

Reply:
Hi,
I don't understand. First, innerp is not checked for null. Second: does the code in 17 differ to 21&head? Else we should change this to an assertion in head, first.

Reply:
Hi Goetz!

`innerp` is the result of 

```
  u_char* get_user_ptr() const {
    assert(_base_addr != nullptr, "Not wrapping any memory");
    return _base_addr + sizeof(GuardHeader);
  }
```

which cannot return null. _base_addr is the result of a malloc and we only ever enter this path if that malloc succeeded. It is also asserted at least twice.

Cheers, Thomas

Reply:
Yes, but the new code checks whether "ptr" is null. That is passed in as an argument.

Reply:
Okay, I just misread the code. Sorry for the confusion, I was too tired.